tools/ExamplesGenerator.ps1

```diff
@@ -6,17 +6,36 @@ Param(
 )
 function Start-Generator {
     Param(
-        $ModulesToGenerate = @()
+        $ModulesToGenerate = @(),
+        [ValidateNotNullOrEmpty()]
+        [string] $GenerationMode = "auto",
+        [string] $GraphCommand = "GetMg-User",
+        [string] $GraphModule = "Users",
+        [string] $ProfilePath = "v1.0",
+        [string] $ManualExternalDocsUrl = "https://learn.microsoft.com/en-us/graph/api/user-get?view=graph-rest-1.0&tabs=powershell"
     )
 
     $GraphMapping = @{
         "v1.0" = "examples\v1.0"
         "beta" = "examples\v1.0-beta"
     }
-    $GraphMapping.Keys | ForEach-Object {
-        $graphProfile = $_
-        Get-FilesByProfile -GraphProfile $graphProfile -GraphProfilePath $GraphMapping[$graphProfile] -ModulesToGenerate $ModulesToGenerate 
+    if ($GenerationMode -eq "auto") {
+        $GraphMapping.Keys | ForEach-Object {
+            $graphProfile = $_
+            Get-FilesByProfile -GraphProfile $graphProfile -GraphProfilePath $GraphMapping[$graphProfile] -ModulesToGenerate $ModulesToGenerate 
+        }
+    }
+    else {
+          
+        $ProfilePathMapping = "examples\v1.0"
+        if($ProfilePath -eq "beta"){
+            $ProfilePathMapping = "examples\v1.0-beta"
+        }
+        $ModulePath = Join-Path $PSScriptRoot "..\src\$GraphModule\$GraphModule\$ProfilePathMapping"
+        Get-ExternalDocs-Url -ManualExternalDocsUrl $ManualExternalDocsUrl -GenerationMode $GenerationMode -GraphProfilePath $ModulePath -Command $GraphCommand -GraphProfile $ProfilePath -Module -$Module
```

Comment:
Please rename `Get-ExternalDocs-Url` function to be compliant with PowerShell naming convention.
```suggestion
        Get-ExternalDocsUrl -ManualExternalDocsUrl $ManualExternalDocsUrl -GenerationMode $GenerationMode -GraphProfilePath $ModulePath -Command $GraphCommand -GraphProfile $ProfilePath -Module -$Module
```